app/todos/todos.module.js

```diff
@@ -20,18 +20,9 @@ angular
             })
             .state({
                 name: 'todos.upsert',
-                url: '/upsert',
+                url: '/upsert/:id',
                 templateUrl: 'todos/upsert/upsert-todo.html',
                 controller: 'UpsertTodoController',
                 controllerAs: 'vm'
```

Comment:
This is one state i.e. a set of  view and controller to add and edit tag. If an id is passed, it becomes an **Edit Todo** form and if it not present, it becomes an **Add Todo** form.